client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -0,0 +1,282 @@
+						fields={ fields }
+						form={ { type: 'regular', fields } }
+						onChange={ ( edits: { domain?: string } ) => {
+							setFormData( ( data ) => ( {
+								...data,
+								...edits,
+								domain: edits.domain?.trim() ?? data.domain,
+							} ) );
+						} }
+					/>
+					<HStack spacing={ 4 } justify="flex-end">
+						<Button variant="tertiary" onClick={ onClose }>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							variant="primary"
+							type="submit"
+							isBusy={ isBusy }
+							disabled={ formData.domain !== siteDomain }
+						>
+							{ __( 'Reset site' ) }
+						</Button>
+					</HStack>
+				</VStack>
+			</form>
+		</VStack>
+	);
+}
+
+export default function SiteResetModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const { data: siteContent } = useQuery< SiteContent >( siteResetContentSummaryQuery( site.ID ) );
+
+	const refreshStatusQuery = {
+		...siteResetStatusQuery( site.ID ),
+		...( site.is_wpcom_atomic && {
+			refetchInterval: ( query: Query< SiteResetStatus > ) => {
+				const { data } = query.state;
+
+				if ( data?.status === 'ready' || data?.status === 'completed' ) {
+					return false;
+				}
+				return 5000;
+			},
+		} ),
+	};
+
+	const { data: resetStatus, refetch: refetchResetStatus } =
+		useQuery< SiteResetStatus >( refreshStatusQuery );
+	const { mutate, isPending: isMutationPending } = useMutation< void, ErrorResponse >(
+		resetSiteMutation( site.ID )
+	);
+
+	const queryClient = useQueryClient();
+
+	const clearQueries = useCallback( () => {
+		queryClient.removeQueries( {
+			queryKey: siteResetContentSummaryQuery( site.ID ).queryKey,
+		} );
+
+		queryClient.removeQueries( {
+			queryKey: siteResetStatusQuery( site.ID ).queryKey,
+		} );
+	}, [ queryClient, site.ID ] );
+
+	const showSuccessNotice = useCallback( () => {
+		createSuccessNotice(
+			sprintf(
+				/* translators: %s: site domain */
+				__( '%s has been reset.' ),
+				site.slug
+			),
+			{ type: 'snackbar' }
+		);
+	}, [ createSuccessNotice, site.slug ] );
+
+	useEffect( () => {
+		if ( resetStatus?.status === 'completed' ) {
+			clearQueries();
+			showSuccessNotice();
+			onClose();
+		}
+	}, [ clearQueries, resetStatus?.status, showSuccessNotice, onClose ] );
+
+	if ( ! siteContent || ! resetStatus ) {
+		return null;
+	}
+
+	const handleReset = () => {
+		if ( isMutationPending ) {
+			return;
+		}
+
+		mutate( undefined, {
+			onSuccess: () => {
+				if ( site.is_wpcom_atomic ) {
+					refetchResetStatus();
+				} else {
+					clearQueries();
+					showSuccessNotice();
+					onClose();
+				}
+			},
+			onError: ( error: ErrorResponse ) => {
+				if ( error.code === 'staging_sites_not_supported' ) {
+					setError( error.message );
+				} else {
+					setError( __( 'The site could not be reset due to an error. Please try again.' ) );
```

Comment:
There's something about "Please try again" that feels off here. Insofar as if we really ask the user to try again, can't we just do it for them? I don't know that "You can try again" is better, and I wonder, can we do without that second sentence entirely?

Reply:
I think we can just use `error.message` regardless of the `error.code`, and fall back to a default message if needed. That way, we can avoid defining a separate ErrorResponse type 🤔

Reply:
Let me double-check the endpoint.

Reply:
Oh yes, that’s a fair point. However, having distinct error messages could help with debugging, since users typically report issues based on what they see. If all the messages look the same, it becomes difficult to trace where the error is coming from 😞

Maybe we should consider updating the message at the endpoint level but that’s something we can iterate on later 🙂